Nightly job is failing for all platforms.

Nightly job has started failing for all platforms. 

The recent merge of PR #29596 seems to be the cause .

For [linux](https://github.com/servo/servo/actions/runs/4664157901/jobs/8256134838) and mac builds, it seems like the secrets need to be [explictly passed to reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow).

```
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

  File "/home/runner/work/servo/servo/python/servo/package_commands.py", line 766, in upload_nightly
    upload_to_s3(platform, package, timestamp)
  File "/home/runner/work/servo/servo/python/servo/package_commands.py", line 616, in upload_to_s3
    (aws_access_key, aws_secret_access_key) = get_s3_secret()
  File "/home/runner/work/servo/servo/python/servo/package_commands.py", line 604, in get_s3_secret
    secret = json.loads(os.environ['S3_UPLOAD_CREDENTIALS'])
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
Error: Process completed with exit code 1.
```


For windows build, I'm not sure what the issue is. The logs seem to indicate issue with file paths
EDIT: the issue could be missing `working-directory: "C:\\a\\servo\\servo"` property on the [upload step](https://github.com/mukilan/servo/blob/master/.github/workflows/windows.yml#L83). 

```
Run python mach upload-nightly windows-msvc --secret-from-environment
  python mach upload-nightly windows-msvc --secret-from-environment
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    RUST_BACKTRACE: 1
    SHELL: /bin/bash
    LAYOUT: layout-2013
    PACKAGE: windows-msvc
    S3_UPLOAD_CREDENTIALS: 
Could not find package for windows-msvc at target\release\msi\Servo.exe
Error: Process completed with exit code 1.
```

cc: @sagudev

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -19,6 +19,7 @@ jobs:
     with:
       layout: '2013'
       upload: true
+    secrets: inherit
 
   upload-mac:
     # This job is only useful when run on upstream servo.
@@ -28,6 +29,7 @@ jobs:
     with:
       layout: '2013'
       upload: true
+    secrets: inherit
 
   upload-linux:
     # This job is only useful when run on upstream servo.
@@ -37,6 +39,7 @@ jobs:
     with:
       layout: '2013'
       upload: true
+    secrets: inherit
 
   upload-linux-2020:
     # This job is only useful when run on upstream servo.
@@ -46,3 +49,4 @@ jobs:
     with:
       layout: '2020'
       upload: true
+    secrets: inherit

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -82,6 +82,7 @@ jobs:
           path: C:\\a\\servo\\servo\\target/release/msi/Servo.exe
       - name: Upload
         if: ${{ inputs.upload }}
+        working-directory: "C:\\a\\${{ github.event.repository.name }}\\${{ github.event.repository.name }}"
         run: python mach upload-nightly windows-msvc --secret-from-environment
         env:
           S3_UPLOAD_CREDENTIALS: ${{ secrets.S3_UPLOAD_CREDENTIALS }}

